sale_exception/models/sale_order_line.py

```diff
@@ -11,8 +11,9 @@ class SaleOrderLine(models.Model):
     _name = "sale.order.line"
 
     exception_ids = fields.Many2many(
-        "exception.rule", string="Exceptions", copy=False, readonly=True
+        related="order_id.exception_ids", string="Exceptions", copy=False, readonly=True
```

Comment:
Why doing this ?
exception_ids should be the exceptions related to the line not to the order. In your case the decoration condition will be True for all SO lines since there is at least one exception on the SO or on a SO line.

Reply:
By the way can you agree with https://github.com/OCA/sale-workflow/pull/2499 and accept this solution ?

Reply:
Hi @glitchov . That's correct, I made a mistake thinking about it. I agree with #2499 . It's more complete and clean than mine :)